### ISSUE TITLE
Notify screen reader users when a link in the editor is disabled

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#31378]
## What problem does this address?

There are blocks that displays pseudo links (`<a href="#block-name-pseudo-link">`) or links where clicking has been disabled
with onClick preventDefault, but there is no notification for screen reader users that the link is disabled.

The reason for having a link element in the editor for these blocks is so that the markup in the editor and front match.

Examples:
NVDA reads the the login/logout block in the editor as:
`Block: Login/out  grouping  clickable   link  Log out`

[Example, Login/out edit.js line 40](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/loginout/edit.js#L40)

## What is your proposed solution?

Identify the blocks that displays but disables actions on **focusable** links, and reconsider what the best solution would be for that block.
If using the link must be disabled, one option would be to add an aria-label that explains that the link is disabled.

- [ ] Login / out
- [ ] Post date
- [ ] Post comments link
- [ ] Post hierarchical terms
- [ ] Post tags

**Comment by joedolson** [comment]
This is an interesting question.  They're there to represent the content, and not intended for interaction, so having the link non-interactive seems totally reasonable to me; but whether a user should be informed of that, specifically, is a different question.

Technically, I'm not sure this is an accessibility issue - visually, the link also presents as if it were a link, but doesn't function as a link for sighted mouse users. Same for sighted keyboard users. So there isn't an unequal presentation of content between different interaction models.

Whether there *should* be presentation showing that the links are disabled is a separate question. I think that there probably should be, but how it should be presented isn't clear to me. It would create a lot of excess noise if it's part of every link individually, but indicating it on the block level feels clumsy to me. What would that look like? "Links in this block are disabled in editing"? 

The common factor, I think, is that any system-generated link (not editable within the editor) is disabled. This seems like something that should be communicable in some way - but maybe it just needs to be documentation. Not sure.

**Comment by carolinan** [comment]
This [article by Scott Ohara about disabling links](https://www.scottohara.me/blog/2021/05/28/disabled-links.html) might help.
Summary: remove href, add `role="link"` and `aria-disabled="true"`.

But if I remember correctly from earlier testing, removing the href means it is no longer styled as a link and the editor and front no longer matches visually.
